dali/base/dadfs.cpp

```diff
@@ -6121,6 +6128,13 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
                 throw MakeStringException(-1, "The value of @numsubfiles (%d) is not equal to the number of SubFile items (%d)!",numSubfiles, subFileCount);
         }
     }
+    virtual bool isRestrictedAccess() override
+    {
+        // This ensures restriction applies even if superfile is unrestricted but subfiles are.
+        // However, this also means that the superfile will show as "Restricted" in ECL Watch and whenever the user tries to unset the flag
+        // it will appear to reset to Restricted.
+        return containsRestrictedSubfile || queryAttributes().getPropBool("@restricted");
```

Comment:
It might get confusing if a subfile implies super is restricted, but also the super has a restricted attribute of it's own..
And if super has restricted flag but subs don't there's nothing to stop the user looking at the super (if that doesn't count as reading?) and then reading the unprotected subs directly.

I think I wouldn't have a separate super restricted attribute.